File: Tests/SwiftKotlinFrameworkTests/Tests/KotlinTokenizer/enums.swift

enum CompassPoint {
    case north
    case south
    case east
    case west
}

private enum Planet {
    case mercury, venus, earth
    case mars, jupiter, saturn, uranus, neptune
}

enum Barcode {
    case upc(Int, Int, Int, Int)
    case qrCode(named: String)
    case empty
}

// Usage
switch enumValue {
case .resetPasswordSendEmail:
    return (category: "ResetPassword", name: "sendEmail", label: nil)
case .paymentSelectorOpen(_, let tenant, _, let option):
    return (category: "PaymentSelector", name: "open", label: "\(tenant.name) - \(option.duration)min")
}

switch exception {
case .qrCode(_):
    if let message = serverMessage {
        trackError(name: name, message: message)
    } else {
        trackError(name: name, message: R.string.localizable.network_error())
    }
default:
    trackError(name: "generic", message: R.string.localizable.generic_error())
}

File: Tests/SwiftKotlinFrameworkTests/Tests/KotlinTokenizer/enums.kt
enum class CompassPoint {
    north,
    south,
    east,
    west
}
private enum class Planet {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune
}
sealed class Barcode {
    data class upc(val v1: Int, val v2: Int, val v3: Int, val v4: Int) : Barcode()
    data class qrCode(val named: String) : Barcode()
    object empty : Barcode()
}
when (enumValue) {
    .resetPasswordSendEmail -> return (category: "ResetPassword", name: "sendEmail", label: null)
    .paymentSelectorOpen -> return (category: "PaymentSelector", name: "open", label: "${tenant.name} - ${option.duration}min")
}
